CRM/Contribute/BAO/Contribution.php

```diff
@@ -5457,4 +5457,37 @@ public static function createProportionalFinancialEntries($entityParams, $lineIt
     }
   }
 
+  /**
+   * Calculate amounts when Financial type for contribution is changed.
+   *
+   * @param array $params
+   *   contribution params
+   * @param float $trxnAmount
+   *   financial trxn total amount
+   * @param float $totalAmount
+   *   Contribution total amount
+   *
+   * @return array
+   */
+  public static function calculateFTChangeAmount(&$params, $trxnAmount, &$totalAmount) {
```

Comment:
The name for this function doesn't seem to properly reflect what the function does, since financial type is not checked.  Maybe calculateAmountsOnChange?